static/app/views/replays/detail/userActionsNavigator.tsx

```diff
@@ -26,17 +26,42 @@ type Props = {
   event: EventTransaction;
 };
 
+type PanelItemCenterProps = {
+  isHovered?: boolean;
+  isSelected?: boolean;
+};
+
 function UserActionsNavigator({event, crumbs}: Props) {
-  const {setCurrentTime} = useReplayContext();
+  const {setCurrentTime, currentHoverTime} = useReplayContext();
   const [currentUserAction, setCurrentUserAction] = useState<Crumb>();
+  const [closestUserAction, setClosestUserAction] = useState<Crumb>();
+
+  const {startTimestamp} = event;
+  const userActionCrumbs = onlyUserActions(transformCrumbs(crumbs));
+
+  useEffect(() => {
+    if (!currentHoverTime || !userActionCrumbs || !startTimestamp) {
+      setClosestUserAction(undefined);
+      return;
+    }
+    // TODO: Find a better way to find the closest user action without using `reduce` method
+    const closestUserActionItem = userActionCrumbs.reduce((prev, curr) => {
+      return Math.abs(
+        relativeTimeInMs(curr.timestamp ?? '', startTimestamp) - currentHoverTime
+      ) <
+        Math.abs(
+          relativeTimeInMs(prev.timestamp ?? '', startTimestamp) - currentHoverTime
+        )
+        ? curr
+        : prev;
+    });
+    setClosestUserAction(closestUserActionItem);
+  }, [currentHoverTime, startTimestamp, userActionCrumbs]);
```

Comment:
The reduce call, or whatever replaces it, would be good to put in a utils function so we can test it directly. It only depends on a list of crumbs and a timestamp, so it's really independent of any react component.